utils/polus-python-template/{{cookiecutter.project_slug}}/src/main.py

```diff
@@ -1,96 +1,165 @@
+    
+    This function should do something, but for now just returns the input.
+
+    Args:
+        input_data: A numpy array.
+        
+    Returns:
+        np.ndarray: Returns the input image.
+    """
+    
+    return input_data
+    
+{# This generates the main function definition -#}
+{#- Required inputs are arguments -#}
+{#- Optional inputs are keyword arguments -#}
+def main({#- Required inputs -#}
+         {% for inp,val in cookiecutter._inputs.items() -%}
+         {%     if val.required -%}
+         {%         if val.type=="boolean" -%}
+         {{ inp }}: bool,
+         {%         elif val.type=="collection" and cookiecutter.use_bfio -%}
+         {{ inp }}: Path,
+         {%         else -%}
+         {{ inp }}: str,
+         {%         endif -%}
+         {%     endif -%}
+         {% endfor -%}
+         {#- Required Outputs (all outputs are required, and should be a Path) -#}
+         {% for inp,val in cookiecutter._outputs.items() -%}
+         {{ inp }}: Path,
+         {% endfor -%}
+         {#- Optional inputs -#}
+         {% for inp,val in cookiecutter._inputs.items() -%}
+         {%     if not val.required -%}
+         {%         if val.type=="boolean" -%}
+         {{ inp }}: typing.Optional[bool] = None,
+         {%         elif val.type=="collection" and cookiecutter.use_bfio -%}
+         {{ inp }}: typing.Options[Path] = None,
+         {%         else -%}
+         {{ inp }}: typing.Optional[str] = None,
+         {%         endif -%}
+         {%     endif -%}
+         {% endfor -%}) -> None:
+    """ Main execution function
+    
+    All functions in your code must have docstrings attached to them, and the
+    docstrings must follow the Google Python Style:
+    https://www.sphinx-doc.org/en/master/usage/extensions/example_google.html
+    """
+    {# Initialize a filepattern object if filepattern is going to be used -#}
+    {%- if cookiecutter.use_filepattern == "True" %}
+    pattern = filePattern if filePattern is not None else '.*'
+    fp = filepattern.FilePattern(
+        {%- for inp,val in cookiecutter._inputs.items() if val.type=='collection' -%}
+        {% if loop.first %}{{ inp }}{% endif %}
+        {%- endfor -%},pattern)
+    
+    for files in fp:
+        # get the first file
+        file = files.pop()
+        
+    {%- else %}
+    files = list({% for inp,val in cookiecutter._inputs.items() if val.type=='collection' -%}
+        {% if loop.first %}{{ inp }}{% endif %}
+        {%- endfor -%}.iterdir())
+    
+    for file in files:
+        
+    {%- endif %}
+    {#- Use bfio if requested #}
+    {%- if cookiecutter.use_bfio == "True" %}
+    {%- filter indent(level2,True) %}
+    
+    logger.info(f'Processing image: {file["file"]}')
+    
+    # Load the input image
+    logger.debug(f'Initializing BioReader for {file["file"]}')
+    with BioReader(file['file']) as br:
+        
+        input_extension = ''.join([s for s in file['file'].suffixes[-2:] if len(s) < 5])
```

Comment:
This seems pretty brittle. Personally I'd do something like this, then take the second element of the array.

```
'filename.ext1.ext2'.split(os.extsep, 1)
['filename', 'ext1.ext2']
```